### PR TITLE
WIP: Replace strcat to omrstr_printf in jclcinit.c

### PR DESCRIPTION
**Pull request by VijiKannan**
Replaced usage of strcat to omrstr_printf to improve performance
Issue: #272

Signed-off-by: Vijayalakshmi Kannnan <k.vijayalakshmi@in.ibm.com>

### CODE REVIEW

File: runtime/jcl/common/jclcinit.c [hunk 1]
```diff
@@ -66,87 +67,85 @@ jint computeFullVersionString(J9JavaVM* vm)
...
-	strcpy(fullversion, "JRE ");
-	strcpy(vminfo, "JRE ");
+        buflen = omrstr_printf(fullversion, sizeof(fullversion), "JRE ");
+        vmlen = omrstr_printf(vminfo, sizeof(vminfo), "JRE ");
 	switch(J2SE_VERSION(vm) & J2SE_VERSION_MASK) {
 	case J2SE_17:
 		if ((J2SE_VERSION(vm) & J2SE_RELEASE_MASK) == J2SE_17) {
-			strcat(fullversion, "1.7.0 ");
-			strcat(vminfo, "1.7.0 ");
+		        buflen += omrstr_printf(fullversion+buflen, sizeof(fullversion), "1.7.0 ");
```
Comment (pshipton) [inline runtime/jcl/common/jclcinit.c]: Please remove the whitespace which was added at the end of most of the omrstr_printf() lines.

File: runtime/jcl/common/jclcinit.c [hunk 2]
```diff
@@ -66,87 +67,85 @@ jint computeFullVersionString(J9JavaVM* vm)
...
 	VMI_ACCESS_FROM_JAVAVM((JavaVM*)vm);
 	PORT_ACCESS_FROM_JAVAVM(vm);
+	OMRPORT_ACCESS_FROM_J9PORT(vm->portLibrary);
 	const char *osarch;
 	const char *osname;
 	const char *vmVersion = NULL;
 	const char *gcVersion = NULL;
-	char fullversion[512];
 	char vminfo[512];
+        char fullversion[512];
```
Comment (pshipton) [inline runtime/jcl/common/jclcinit.c]: Please use tabs in all the modified code, to be consistent with the rest of the file.

File: runtime/jcl/common/jclcinit.c (hunk 1 above)
Comment (pshipton) [reply runtime/jcl/common/jclcinit.c]: The correct length is sizeof(fullversion)-buflen. Similarly for vminfo.

**Comment by pshipton** [comment]
Please change the commit comment and Pull Request comment to contain "Fixes #272" or "Closes #272" etc. in order to auto-close the issue when this PR is merged. 

https://help.github.com/articles/closing-issues-using-keywords/

**Comment by keithc-ca** [comment]
Using `omrstr_printf` with computed destinations is an improvement, but my hope was that the result would be that `computeFullVersionString` contains exactly two calls to `omrstr_printf` (one each for `fullversion` and `vminfo`).

**Comment by VijiKannan** [comment]
Thanks for the comments, I  am currently modifying the code. I will post the updated version soon

**Comment by keithc-ca** [comment]
Closing: the issue was addressed by #648.

**Pull request closed without merging**